### PR TITLE
Adding support for regular expression matching for values of type Chars

### PR DESCRIPTION
**Pull request by WamanAvadhani**
This adds support for regular expressions for matching the values of type Chars.
Example: Chars   *   <field_name>    REGEXP:<reg_ex>
The same field could also be used as header_filter to filter the messages using regular expressions. #50

### CODE REVIEW

**Comment by samratashok87** [comment]
Good Feature to have

File: atest/header_filter.robot [hunk 1]
```diff
@@ -24,11 +62,12 @@ Setup up server for test
 
 Define a protocol with string field in header
     New protocol    StringInHeader
-    Chars     *     string_field 
+    Uint      1     integer_field
+    Chars     *     string_field
```
Comment (jussimalinen) [inline atest/header_filter.robot]: Maybe add a terminator for this field? We could then try sending two messages and then receiving the correct one with the filter?

**Comment by jussimalinen** [comment]
Excellent, there is one small test change I would like (in notes), but otherwise it is ready.

**Comment by jussimalinen** [comment]
Thanks! Maybe better if we dont use dot in the values if the terminator is the dot? Can be a bit confusing?

**Comment by jussimalinen** [comment]
Thanks!

**Pull request merged**